darwin/future/meta/queries/dataset.py

```diff
@@ -0,0 +1,59 @@
+from typing import List
+
+from darwin.future.core.client import Client
+from darwin.future.core.datasets.list_datasets import list_datasets
+from darwin.future.core.types.query import Modifier, Param, Query, QueryFilter
+from darwin.future.data_objects.dataset import Dataset
+from darwin.future.data_objects.release import ReleaseList
+
+
+class DatasetQuery(Query[Dataset]):
+    """
+    DatasetQuery object with methods to manage filters, retrieve data, and execute
+    filters
+
+    Methods
+    -------
+
+    where: Adds a filter to the query
+    collect: Executes the query and returns the filtered data
+    """
+
+    def where(self, param: Param) -> "DatasetQuery":
+        filter = QueryFilter.parse_obj(param)
+        query = self + filter
+
+        return DatasetQuery(query.filters)
+
+    def collect(self, client: Client) -> List[Dataset]:
+        datasets, exceptions = list_datasets(client)
+        if exceptions:
+            # TODO: print and or raise exceptions, tbd how we want to handle this
```

Comment:
We should definitely have a discussion about exceptions soon, figure out where we want to handle them, collecting exceptions into lists, logging (somewhat seperate but related)

Reply:
Yeah, so the issue we have is whether we are withing a CLI or SDK context, I suppose.  We could maybe add `console` member to `client` so that CLI usage would be detectable, or we could use `sys` to detect the call stack, but that may be less x-plat compatible.